sql: DROP after restart hangs

The following tests appear to have failed:

[#237218](https://teamcity.cockroachdb.com/viewLog.html?buildId=237218):

```
--- FAIL: acceptance/TestDockerCLI (157.590s)
test_log_scope.go:79: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestDockerCLI078706760
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestDockerCLI/test_server_restart.tcl (37.030s)
cli_test.go:73: non-zero exit code: 1

```


(dup of #15398)

--- a/pkg/cli/interactive_tests/common.tcl
+++ b/pkg/cli/interactive_tests/common.tcl
@@ -44,5 +44,5 @@ proc start_server {argv} {
     system "mkfifo pid_fifo || true; $argv start --insecure --pid-file=pid_fifo --background >>stdout.log 2>>stderr.log & cat pid_fifo > server_pid"
 }
 proc stop_server {argv} {
-    system "set -e; if kill -CONT `cat server_pid`; then $argv quit || true & sleep 1; kill -9 `cat server_pid` || true; else $argv quit || true; fi"
+    system "$argv quit"
 }

